app/models/order_detail.rb

```diff
@@ -942,7 +942,7 @@ def clear_statement
 
   def update_fulfilled_at_on_resolve
     if problem_changed? && !problem_order?
-      self.fulfilled_at = time_data.actual_end_at
+      self.fulfilled_at = time_data.actual_end_at if time_data.actual_end_at
```

Comment:
So this is a callback that happens anytime we save an order detail, not just for problem orders. Will we never have a case where `time_data.actual_end_at` changes, but the previous value is not nil?

```
before_save :set_problem_order
  def set_problem_order
    self.problem = problem_description_key.present?
    update_fulfilled_at_on_resolve if time_data.present?
    true # problem might be false; we need the callback chain to continue
  end
```